src/webgpu/util/texture/texelData.ts

```diff
@@ -322,170 +155,364 @@ class TexelDataRepresentationImpl implements TexelDataRepresentation {
+    'r8uint':                makeIntegerInfo(      kR,  8, { signed: false }),
+    'r8sint':                makeIntegerInfo(      kR,  8, { signed:  true }),
+    'r16uint':               makeIntegerInfo(      kR, 16, { signed: false }),
+    'r16sint':               makeIntegerInfo(      kR, 16, { signed:  true }),
+    'r16float':              makeFloatInfo(        kR, 16),
+    'rg8unorm':              makeNormalizedInfo(  kRG,  8, { signed: false, sRGB: false }),
+    'rg8snorm':              makeNormalizedInfo(  kRG,  8, { signed:  true, sRGB: false }),
+    'rg8uint':               makeIntegerInfo(     kRG,  8, { signed: false }),
+    'rg8sint':               makeIntegerInfo(     kRG,  8, { signed:  true }),
+    'r32uint':               makeIntegerInfo(      kR, 32, { signed: false }),
+    'r32sint':               makeIntegerInfo(      kR, 32, { signed:  true }),
+    'r32float':              makeFloatInfo(        kR, 32),
+    'rg16uint':              makeIntegerInfo(     kRG, 16, { signed: false }),
+    'rg16sint':              makeIntegerInfo(     kRG, 16, { signed:  true }),
+    'rg16float':             makeFloatInfo(       kRG, 16),
+    'rgba8unorm':            makeNormalizedInfo(kRGBA,  8, { signed: false, sRGB: false }),
+    'rgba8unorm-srgb':       makeNormalizedInfo(kRGBA,  8, { signed: false, sRGB:  true }),
+    'rgba8snorm':            makeNormalizedInfo(kRGBA,  8, { signed:  true, sRGB: false }),
+    'rgba8uint':             makeIntegerInfo(   kRGBA,  8, { signed: false }),
+    'rgba8sint':             makeIntegerInfo(   kRGBA,  8, { signed:  true }),
+    'bgra8unorm':            makeNormalizedInfo(kBGRA,  8, { signed: false, sRGB: false }),
+    'bgra8unorm-srgb':       makeNormalizedInfo(kBGRA,  8, { signed: false, sRGB:  true }),
+    'rg32uint':              makeIntegerInfo(     kRG, 32, { signed: false }),
+    'rg32sint':              makeIntegerInfo(     kRG, 32, { signed:  true }),
+    'rg32float':             makeFloatInfo(       kRG, 32),
+    'rgba16uint':            makeIntegerInfo(   kRGBA, 16, { signed: false }),
+    'rgba16sint':            makeIntegerInfo(   kRGBA, 16, { signed:  true }),
+    'rgba16float':           makeFloatInfo(     kRGBA, 16),
+    'rgba32uint':            makeIntegerInfo(   kRGBA, 32, { signed: false }),
+    'rgba32sint':            makeIntegerInfo(   kRGBA, 32, { signed:  true }),
+    'rgba32float':           makeFloatInfo(     kRGBA, 32),
+  },
+  ...{
+    rgb10a2unorm: {
+      componentOrder: kRGBA,
+      componentInfo: {
+        [TexelComponent.R]: { dataType: 'unorm', bitLength: 10 },
+        [TexelComponent.G]: { dataType: 'unorm', bitLength: 10 },
+        [TexelComponent.B]: { dataType: 'unorm', bitLength: 10 },
+        [TexelComponent.A]: { dataType: 'unorm', bitLength: 2 },
+      },
+      encode: components => {
+        components = applyEach(
+          (n: number) => (floatAsNormalizedInteger(n, 10, false), n),
```

Comment:
This is discarding the result of floatAsNormalizedInteger? I don't get why. Are the packed formats special in that they're encoded later?

Reply:
hm it's not - this is doing something akin to:

components = Object.assign({}, components, {
  R: floatAsNormalized(components.R),
  G: floatAsNormalized(components.G),
  B: floatAsNormalized(components.B),
});
components = Object.assign({}, components, {
  A: floatAsNormalized(components.A),
});

Reply:
but yes as mentioned below maybe it would be more clear to just repeat the code

Reply:
but shouldn't `(n: number) => (floatAsNormalizedInteger(n, 10, false), n)` be an identity function (sequence operator `,`), except for side effects?

Reply:
Done. The discarding was a bug.